flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/Catalog.java

```diff
@@ -206,6 +206,22 @@ void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
     void alterDatabase(String name, CatalogDatabase newDatabase, boolean ignoreIfNotExists)
             throws DatabaseNotExistException, CatalogException;
 
+    /**
+     * Rename an existing database.
+     *
+     * @param name name of the database to be renamed
+     * @param newDatabaseName new name of the database
+     * @param ignoreIfNotExists flag to specify behavior when database does not exist: if set to
+     *     false, throw an exception, if set to true, do nothing
+     * @throws DatabaseNotExistException if the database does not exist
+     * @throws DatabaseAlreadyExistException if the database with newDatabaseName already exists
+     * @throws CatalogException in case of any runtime exception
```

Comment:
Probably it is questionable and should be rephrased....
because for example in implementation it is used `org.apache.flink.table.catalog.GenericInMemoryCatalog#databaseExists` which could fail with `java.lang.IllegalArgumentException` which is `RuntimeException`. So it will not be caught and rethrown  with `CatalogException`

Reply:
Hi @snuyanzin , thanks for your review!
I saw [renameTable](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/Catalog.java#L271) follows the same pattern for description, cause in its implementation it calls `checkArgument` which could potentially throw `java.lang.IllegalArgumentException`.
Do you think we should update the function signature to something like this?
```
    /**
     * Rename an existing table or view.
     *
     * @param tablePath Path of the table or view to be renamed
     * @param newTableName the new name of the table or view
     * @param ignoreIfNotExists Flag to specify behavior when the table or view does not exist: if
     *     set to false, throw an exception, if set to true, do nothing.
     * @throws TableNotExistException if the table does not exist
     * @throws IllegalArgumentException when provided argument is not valid
     * @throws CatalogException in case of any other runtime exception
     */
    void renameTable(ObjectPath tablePath, String newTableName, boolean ignoreIfNotExists)
            throws TableNotExistException, TableAlreadyExistException, IllegalArgumentException, CatalogException;
```

Reply:
Yes probably you right, it wouldn't make sense to do it only for one method...
Probably makes sense to do in a separate PR...